python/ml4ir/applications/ranking/tests/test_feature_layer.py

```diff
@@ -555,13 +599,98 @@ def test_categorical_indicator_with_vocabulary_file_without_ids(self):
 
         # Strings domain_0 and domain_2 should NOT result in the same one-hot vector because they use a default one-to-one vocabulary mapping
-        assert not tf.reduce_all(tf.equal(categorical_one_hot[0], categorical_one_hot[3]))
-        assert not tf.reduce_all(tf.equal(categorical_one_hot[3], categorical_one_hot[4]))
+        assert not tf.reduce_all(
+            tf.equal(categorical_one_hot[0], categorical_one_hot[3]))
+        assert not tf.reduce_all(
+            tf.equal(categorical_one_hot[3], categorical_one_hot[4]))
+
+    def test_global_1d_pooling(self):
+        """
+        Unit test the global 1D pooling feature function on sequence features
+
+        Checks the right output shapes produced and the values generated
+        """
+        feature_tensor = tf.reshape(
+            tf.constant(range(30), dtype=tf.float32),
+            (2, 5, 3))
+        pooled_tensor = sequence_fns.global_1d_pooling(
+            feature_tensor=feature_tensor,
+            feature_info={
+                "name": "f",
+                "feature_layer_info": {
+                    "args": {
+                        "fns": [
+                            "sum",
+                            "mean",
+                            "max",
+                            "min",
+                            "count_nonzero"
+                        ]
+                    }
+                }
+            },
+            file_io=None
+        )
+
+        assert pooled_tensor.shape == (2, 5, 5)
+        assert (pooled_tensor.numpy() == [[[3.,  1.,  2.,  0.,  2.],
+                                           [12.,  4.,  5.,  3.,  3.],
+                                           [21.,  7.,  8.,  6.,  3.],
+                                           [30., 10., 11.,  9.,  3.],
+                                           [39., 13., 14., 12.,  3.]],
+
+                                          [[48., 16., 17., 15.,  3.],
+                                           [57., 19., 20., 18.,  3.],
+                                           [66., 22., 23., 21.,  3.],
+                                           [75., 25., 26., 24.,  3.],
+                                           [84., 28., 29., 27.,  3.]]]).all()
+
+        # Test function with padded values
+        # Here, we mask all the even values
+        padded_val = -1.
+        feature_tensor_with_mask = tf.where(
+            feature_tensor % 2 == 0, feature_tensor, padded_val)
+        pooled_tensor_with_mask = sequence_fns.global_1d_pooling(
+            feature_tensor=feature_tensor_with_mask,
+            feature_info={
+                "name": "f",
+                "feature_layer_info": {
+                    "args": {
+                        "fns": [
+                            "sum",
+                            "mean",
+                            "max",
+                            "min",
+                            "count_nonzero"
```

Comment:
Just another possible test case maybe. If non of the pooling function is passed or only non-supported pooling functions are passed then the call to the pooling function would crash, no?

Reply:
Yeah, what do you think we can do if no pooling function is passed? Ideally, that cannot be the case. But I can throw an appropriate error if that helps.